cider-velox/test/AggWithRandomDataTest.cpp

```diff
@@ -235,6 +289,257 @@ TEST_F(AggWithRandomDataTest, AVG_Partial_Test) {
   CHECK_EQ(resCount, 10);
 }
 
+TEST_F(AggWithRandomDataTest, AVG_Partial_Arrow_NonGroupBy_Test) {
+  GTEST_SKIP();
+  auto veloxPlan =
+      PlanBuilder()
+          .values(vectors)
+          .project({"l_linenumber"})
+          .aggregation(
+              {}, {"avg(l_linenumber)"}, {}, core::AggregationNode::Step::kPartial, false)
+          .planNode();
+
+  auto substraitPlan =
+      v2SPlanFragmentConvertor_->toSubstraitPlan(veloxPlan, veloxPlan->sources()[0]);
+
+  auto ciderCompileModule =
+      CiderCompileModule::Make(std::make_shared<CiderDefaultAllocator>());
+  auto result = ciderCompileModule->compile(substraitPlan);
+
+  auto outputSchema = result->getOutputCiderTableSchema();
+  auto numCols = outputSchema->getColumnCount();
+  CHECK_EQ(numCols, 1);
+  auto outputHints = outputSchema->getColHints();
+  CHECK_EQ(outputHints[0], ColumnHint::PartialAVG);
+  auto& outputType = outputSchema->getColumnTypeById(0);
+  CHECK_EQ(outputType.has_struct_(), true);
+  CHECK_EQ(outputType.struct_().types()[0].kind_case(),
+           ::substrait::Type::KindCase::kFp64);
+  CHECK_EQ(outputType.struct_().types()[1].kind_case(),
+           ::substrait::Type::KindCase::kI64);
+
+  auto ciderRuntimeModule = std::make_shared<CiderRuntimeModule>(result);
+
+  int rows = 10;
+  std::vector<int64_t> colId0(rows);
+  std::vector<int32_t> colId1(rows);
+  std::vector<double> colId2(rows);
+  std::vector<double> colId3(rows);
+  std::vector<double> colId4(rows);
+  for (int i = 0; i < rows; i++) {
+    colId0[i] = i;
+    colId1[i] = i;
+    colId2[i] = i + 0.1;
+    colId3[i] = i + 0.1;
+    colId4[i] = i + 0.1;
+  }
+
+  ArrowArrayBuilder builder;
```

Comment:
can we avoid using ArrowArrayBuilder and try velox::RowVector -> DataConvertor -> CiderBatch path?